### DISCUSSION TITLE
isomorphic package?

### DISCUSSION THREAD

**Discussion by fanckush** [clerk/javascript#1765] [Support request - DO NOT OPEN HERE] [answered]
## Package

- [x] `@clerk/clerk-js`
- [x] other

## Description

Are there any plans to provide an universal isomorphic package that works both on server side and client side seamlessly? I believe functionality is already embedded in `@clerk/nextjs` but for something projcets that don't use react and have SSR it's pretty challenging to use clerk on both the server and the client.

Also, is it currently possible to use `clerk/nextjs`in a non react project and just import the isomorphic parts?

Currently, i'm using `clerk-js` and i'm not sure how to proceedu

**Comment by jescalan** [comment, marked as answer]
This is not currently on our roadmap. The main issue here is that every framework/setup has different ways that it uses both the client and the server, which is why we have framework-specific packages like the nextjs one. I don't think it would be possible for us to provide a universal isomorphic package that worked with every setup.

If you tell us a little more about how you are trying to set it up, that would be helpful!
